x-pack/plugins/security_solution/server/lib/detection_engine/rule_monitoring/logic/rule_execution_log/create_rule_execution_summary.ts

```diff
@@ -18,36 +18,45 @@ import type { RuleParams } from '../../../rule_schema';
 export const createRuleExecutionSummary = (
   rule: SanitizedRule<RuleParams> | ResolvedSanitizedRule<RuleParams>
 ): RuleExecutionSummary | null => {
+  if (!rule.monitoring) {
+    // This object must be always initialized. If it's not, we say
+    // "ok, we don't have any data about execution of this rule".
+    // We don't throw here because we want to still be able to return
+    // a rule from the API even if its execution summary can't be created.
+    return null;
+  }
```

Comment:
@XavierM Is there a reason why it's typed as nullable?

Reply:
I think that's because at the first time the rule is running the monitoring can be null

Reply:
it can be undefined because we only share this attribute in internal API and when it is public we hide it. Not ideal!

Reply:
This should not happen technically because of this https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/rules_client/methods/create.ts#L153 do you agree? maybe we can overwrite the type here. I will leave this file alone for now. I think the problem was only in the enable.

Reply:
TODO @banderror: change to assertion using `invariant`.